packages/grafana-data/src/types/logsVolume.ts

```diff
@@ -0,0 +1,19 @@
+import { DataQuery } from './query';
+import { DataQueryRequest, DataQueryResponse } from './datasource';
+import { Observable } from 'rxjs';
+
+/**
+ * TODO: This should be added to ./logs.ts but because of cross reference between ./datasource.ts and ./logs.ts it can
+ * be done only after decoupling "logs" from "datasource" (https://github.com/grafana/grafana/pull/39536)
+ *
+ * @internal
+ */
+export interface DataSourceWithLogsVolumeSupport<TQuery extends DataQuery> {
+  getLogsVolumeDataProvider(request: DataQueryRequest<TQuery>): Observable<DataQueryResponse> | undefined;
```

Comment:
@aocenas 

> Nitpick but the panel size could be smaller before we load data, having just a single button in a panel with height 150 does not look ideal.

@ivanahuckova:

> I agree with what @aocenas said - that area with the button to fetch logs volume takes too much place for just being button.

This was to prevent UI from jumping especially when auto-loading is enabled. It may happen that logs volume is still loading while the user reads logs below. Once the volume is loaded all log entries would move down a bit if the panel size was not fixed.  What do you think @Ijin08 @davkal?

Reply:
This will be revised in #39856.